### PR TITLE
etcdctlv3: better watch command

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony

### CODE REVIEW

File: etcdctlv3/README.md [hunk 1]
```diff
@@ -108,3 +108,41 @@ OK
 1
 ./etcdctl range foo
 ```
+
+
+### Watch [options] [key or prefix]
+
+Watch watches on keys or prefixes. Watch command never ends until it meets an error or user kills it.
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: The watch command runs until it encounters an error or is terminated by the user.

File: etcdctlv3/command/watch_command.go [hunk 2]
```diff
@@ -24,20 +24,67 @@ import (
...
+)
+
 // NewWatchCommand returns the cobra command for "watch".
 func NewWatchCommand() *cobra.Command {
-	return &cobra.Command{
-		Use:   "watch",
-		Short: "Watch watches the events happening or happened.",
+	cmd := &cobra.Command{
+		Use:   "watch [key or prefix]",
+		Short: "Watch watches on keys or prefixes",
```
Comment (heyitsanthony) [inline etcdctlv3/command/watch_command.go]: streams events on keys or prefixes?

**Comment by xiang90** [comment]
@heyitsanthony PTAL.

File: etcdctlv3/README.md [hunk 3]
```diff
@@ -108,3 +108,41 @@ OK
...
+
+#### Options
+
+- hex -- print out key and value as hex encode string
+
+- i -- begins an interactive watch session
+
+- prefix -- watch on a prefix if prefix is set.
+
+- rev -- the revision to start watching. Rev can be specified to watch previously happened event.
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: s/Rev can be.../Specifying a revision is useful for observing past events./

**Comment by heyitsanthony** [comment]
lgtm aside from last nit

**Pull request merged**